### PR TITLE
Feature/client/filter sidebar

### LINKED ISSUE #69: Client/Filter-Sidebar: Implement UI for complex filter system

### PR DESCRIPTION
**Pull request by perlaescano**
## Description

Added a new  `FilterSideBar` component, that enables users to filter map locations first by location type and then by parameters specific to each location type and reset filter.
## What's in this change?

- **Added `FilterSideBar` component:** 
  - Allows users to filter locations based on the type (e.g., playground, beach, restroom).
  - Dynamically adjusts filter options based on the selected location type, showing only relevant parameters (e.g., accessibility for playgrounds, ADA status for subway stops).
  - Includes a reset option to clear all selected filters and display the full map of locations.

- [x] Primary filter by location type (e.g., playground, beach, restroom).
- [x] Secondary filter options relevant to the selected location type (e.g., sensory-friendly for playgrounds, ADA status for subway stops).
- [x] Reset functionality to clear all filters.

## Testing changes

- Verified that `FilterSideBar` opens and provides an initial filter by location type.
- Confirmed that after selecting a location type, only relevant parameters appear for further filtering.
- Tested filter application to ensure only relevant locations are shown on the map.
- Checked reset functionality to ensure it removes all filters and restores the full location list.
- Verified correct icon display across different types of locations on the map.

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Good job!

File: client/src/pages/Home.jsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import React, { useState, useEffect, useRef } from 'react';
 import '../style/Home.css';
 import NavBar from '../components/NavBar.jsx';
-import MapComponent from '../components/MapComponent'; 
+import MapComponent from '../components/MapComponent';
+import FilterSideBar from '../components/FilterSidebar.jsx';
```
Comment (Tiffany-Zhu5303) [inline client/src/pages/Home.jsx]: Make sure to correct the typo here from `.../components/FilterSidebar,jsx` to `../components/FilterSideBar.jsx` (case sensitivity for the b in Sidebar to be capitalized)

**Review by Tiffany-Zhu5303 — COMMENTED** [review]
Everything looks good but make sure the typo in Home.jsx on line 5 for the FIlterSideBar is updated before merging. 

Other than that, everything looks good but would like to note that when the nearby locations option is unchecked, the reset filter button takes a while to reset. Maybe consider a loading indicator for next sprint.

File: client/src/components/FilterSideBar.jsx
Comment (Tiffany-Zhu5303) [inline client/src/components/FilterSideBar.jsx:1]: this file name needs to be lowercased too (FilterSidebar.jsx instead of FilterSideBar.jsx)

**Review by Tiffany-Zhu5303 — APPROVED** [review]
Looks good!

**Pull request merged**